File: SDKs/amazonaws.com/codeguru-reviewer/2019-09-19/java/lib/src/main/java/openapisdk/models/shared/CommitDiffSourceCodeType.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * CommitDiffSourceCodeType
 *  A type of <a href="https://docs.aws.amazon.com/codeguru/latest/reviewer-api/API_SourceCodeType"> <code>SourceCodeType</code> </a> that specifies the commit diff for a pull request on an associated repository. The <code>SourceCommit</code> and <code>DestinationCommit</code> fields are required to do a pull request code review. 
**/
public class CommitDiffSourceCodeType {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("DestinationCommit")
    public String destinationCommit;
    public CommitDiffSourceCodeType withDestinationCommit(String destinationCommit) {
        this.destinationCommit = destinationCommit;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("MergeBaseCommit")
    public String mergeBaseCommit;
    public CommitDiffSourceCodeType withMergeBaseCommit(String mergeBaseCommit) {
        this.mergeBaseCommit = mergeBaseCommit;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("SourceCommit")
    public String sourceCommit;
    public CommitDiffSourceCodeType withSourceCommit(String sourceCommit) {
        this.sourceCommit = sourceCommit;
        return this;
    }
}

File: SDKs/amazonaws.com/codeguru-reviewer/2019-09-19/python/src/sdk/models/shared/commitdiffsourcecodetype.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class CommitDiffSourceCodeType:
    r"""CommitDiffSourceCodeType
     A type of <a href=\"https://docs.aws.amazon.com/codeguru/latest/reviewer-api/API_SourceCodeType\"> <code>SourceCodeType</code> </a> that specifies the commit diff for a pull request on an associated repository. The <code>SourceCommit</code> and <code>DestinationCommit</code> fields are required to do a pull request code review. 
    """
    
    destination_commit: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('DestinationCommit') }})
    merge_base_commit: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('MergeBaseCommit') }})
    source_commit: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('SourceCommit') }})
    
